### PR TITLE
feat!: Update flame_audio to AP 1.0.0

### PR DESCRIPTION
**Pull request by luanpotter**
# Description

## Checklist

- [x] The title of my PR starts with a [Conventional Commit] prefix (`fix:`, `feat:`, `docs:` etc).
- [x] I have read the [Contributor Guide] and followed the process outlined for submitting PRs.
- [x] I have updated/added tests for ALL new/updated/fixed functionality.
- [x] I have updated/added relevant documentation in `docs` and added dartdoc comments with `///`.
- [x] I have updated/added relevant examples in `examples`.

## Breaking Change

- [x] Yes, this is a breaking change.
- [ ] No, this is *not* a breaking change.

### Migration instructions

Basically follow [AP's 1.0.0 migration guide](https://github.com/bluefireteam/audioplayers/blob/main/migration_guide.md). Flame Audio APIs were updated to match the new design philosophies.

[issue database]: https://github.com/flame-engine/flame/issues
[Contributor Guide]: https://github.com/flame-engine/flame/blob/main/CONTRIBUTING.md
[Flame Style Guide]: https://github.com/flame-engine/flame/blob/main/STYLEGUIDE.md
[Conventional Commit]: https://conventionalcommits.org

### CODE REVIEW

File: packages/flame/example/lib/main.dart [hunk 1]
```diff
@@ -26,6 +26,7 @@ class MyGame extends FlameGame with HasTappables {
   @override
   void onTapUp(int id, TapUpInfo info) {
     super.onTapUp(id, info);
+
```
Comment (luanpotter) [inline packages/flame/example/lib/main.dart:29]:
```suggestion
```

**Review by spydon — APPROVED** [review]
Needs some doc updates right?

**Comment by luanpotter** [comment]
Possibly

File: doc/flame_audio/audio.md [hunk 2]
```diff
@@ -63,13 +59,29 @@ You can use [the `Bgm` class](bgm.md) (via `FlameAudio.bgm`) to play looping bac
 tracks. The `Bgm` class lets Flame automatically manage the pausing and resuming of background music
 tracks when the game is backgrounded or comes back to the foreground.
 
+You can use [the `AudioPool` class](audio_pool.md) if you want to fire quick sound effects on a very
```
Comment (spydon) [inline doc/flame_audio/audio.md:62]:
```suggestion
You can use [the `AudioPool` class](audio_pool.md) if you want to fire quick sound effects in a very
```

File: doc/flame_audio/audio.md [hunk 3]
```diff
@@ -63,13 +59,29 @@ You can use [the `Bgm` class](bgm.md) (via `FlameAudio.bgm`) to play looping bac
...
+efficient manner. `AudioPool` will keep a pool of `AudioPlayer`s preloaded with a given sound, and
+allow you to fire them very fast in quick succession.
```
Comment (spydon) [inline doc/flame_audio/audio.md:64]:
```suggestion
allow you to play them very fast in quick succession. 
```

File: doc/flame_audio/audio.md [hunk 4]
```diff
@@ -63,13 +59,29 @@ You can use [the `Bgm` class](bgm.md) (via `FlameAudio.bgm`) to play looping bac
...
+Both the `play` and `loop` methods return an instance of an `AudioPlayer` from the
+[audioplayers](https://github.com/bluefireteam/audioplayers) lib, that allows you to stop, pause and
+configure other parameters.
+
+In fact you can always use `AudioPlayer`s directly to gain full control over how your audio is played
+-- the `FlameAudio` class is just a wrapper for common functionality.
+
+## Caching
+
+You can pre-load your audios. Audios need to be stored in the memory the first time they
```
Comment (spydon) [inline doc/flame_audio/audio.md:84]: Audios sound a bit strange, audio files, or audio clips maybe?

File: doc/flame_audio/audio_pool.md [hunk 5]
```diff
@@ -0,0 +1,15 @@
+# AudioPool
+
+An AudioPool is a provider of AudioPlayers that leaves them pre-loaded with
+local assets to minimize delays.
```
Comment (spydon) [inline doc/flame_audio/audio_pool.md:4]:
```suggestion
An AudioPool is a provider of AudioPlayers that are pre-loaded with
local assets to minimize delays.
```

File: doc/flame_audio/audio_pool.md [hunk 6]
```diff
@@ -0,0 +1,15 @@
...
+
+A single Audio Pool always plays the same sound, probably a quick sound
```
Comment (spydon) [inline doc/flame_audio/audio_pool.md:6]:
```suggestion
A single AudioPool always plays the same sound, usually a quick sound
```

File: doc/flame_audio/audio_pool.md [hunk 7]
```diff
@@ -0,0 +1,15 @@
...
+effect, like a laser shooting from your ship or a jump sound for your
+platformer.
+
+The advantage of using Audio Pool is that by configuring a minimum
+(starting) size, and a maximum size, the pool will create and preload
+some players, and allow them to be re-used many times.
+
+You can use the helper method `FlameAudio.createPool` to create Audio
+Pool instances using the same global `FlameAudio.audioCache`.
```
Comment (spydon) [inline doc/flame_audio/audio_pool.md:15]:
```suggestion
You can use the helper method `FlameAudio.createPool` to create AudioPool
instances using the same global `FlameAudio.audioCache`.
```

File: packages/flame_audio/example/lib/main.dart [hunk 8]
```diff
@@ -29,7 +29,11 @@ class AudioGame extends FlameGame with TapDetector {
 
   @override
   Future<void> onLoad() async {
-    pool = await AudioPool.create('fire_2.mp3', minPlayers: 3, maxPlayers: 4);
+    pool = await FlameAudio.createPool(
+      'sfx/fire_2.mp3',
```
Comment (spydon) [inline packages/flame_audio/example/lib/main.dart:33]: `sfx` is no longer default right?

File: packages/flame_audio/lib/audio_pool.dart [hunk 9]
```diff
@@ -6,54 +6,47 @@ import 'package:synchronized/synchronized.dart';
 /// Represents a function that can stop an audio playing.
 typedef Stoppable = void Function();
 
-/// An AudioPool is a provider of AudioPlayers that leaves them pre-loaded to
-/// minimize delays.
+/// An AudioPool is a provider of AudioPlayers that leaves them pre-loaded with
+/// local assets to minimize delays.
```
Comment (spydon) [inline packages/flame_audio/lib/audio_pool.dart:10]:
```suggestion
/// An AudioPool is a provider of AudioPlayers that are pre-loaded with
/// local assets to minimize delays.
```

**Review by spydon — APPROVED** [review]
Lgtm, just some doc comments

File: packages/flame_audio/example/lib/main.dart (hunk 8 above)
Comment (luanpotter) [reply packages/flame_audio/example/lib/main.dart:33]: no, I thought that was just complicating things. now they all use the same cache

File: doc/flame_audio/audio.md (hunk 4 above)
Comment (luanpotter) [reply doc/flame_audio/audio.md:84]:
```suggestion
You can pre-load your assets. Audios need to be stored in the memory the first time they
```

**Review by spydon — APPROVED** [review]
Lgtm!

**Comment by spydon** [comment]
Ooh, would actually be awesome with at short example in the examples directory too.

File: doc/flame_audio/audio.md [hunk 10]
```diff
@@ -12,11 +12,10 @@ dependencies:
 
 The latest version can be found on [pub.dev](https://pub.dev/packages/flame_audio/install).
 
-After installing the `flame_audio` package you can add audio files in the assets section of your
+After installing the `flame_audio` package, you can add audio files in the assets section of your
 `pubspec.yaml` file. Make sure that the audio files exists in the paths that you provide.
 
-The default directory for `FlameAudio` is `assets/audio` (which can be changed) and for `AudioPool`
-the default directory is `assets/audio/sfx`.
+The default directory for `FlameAudio` is `assets/audio` (which can be changed).
```
Comment (erickzanardo) [inline doc/flame_audio/audio.md:18]: Can we explain how it can be changed?

Comment (luanpotter) [inline doc/flame_audio/audio.md:18]:
```suggestion
The default directory for `FlameAudio` is `assets/audio` (which can be changed by providing your own instance of `AudioCache`).
```

**Comment by spydon** [comment]
> I already provide an example underr package/flame_audio/example. Do you think we need to add one on the dashbook examples app too? If so should it be the same? Or something different?

The same one I think would be fine, I think our users more easily find the dashbook examples.
